### PR TITLE
Added FastAPI server that displays Anime content

### PR DESCRIPTION
**Pull request by Max2288**

### CODE REVIEW

**Comment by siriusdevs** [comment]
какие модули устанавливать? какая база нужна? какие таблицы? чего делать вообще? почему репозиторий без ридми?

**Review by siriusdevs — CHANGES_REQUESTED** [review]
команду на запуск контейнера с монгой, requrements.txt для pip, ddl файл для таблиц в базе или описание таблиц в базе и команды для создания, ридми для описания того, что с этим делать и что оно умеет делать

File: fastapi/config.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+load_dotenv()
+
+DBNAME = getenv('DBNAME')
+COLNAME = getenv('COLNAME')
+TOKEN = getenv('TOKEN')
+HOST = getenv('HOST')
+PORT = getenv('PORT')
+try:
+  APP_PORT = int(getenv('APP_PORT'))
+except ValueError:
```
Comment (siriusdevs) [inline fastapi/config.py]: если переменной нет в .env файле или она называется не так, это будет None. В таком случае int() выдаст TypeError, а не ValueError, и код упадёт с ошибкой

**Review by siriusdevs — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -1 +1,96 @@
...
+
+
+# How run
+Add .env file with your data
+
+Go to folder with project and run commands:
+
+_$ docker run -d -p 27018:27017 mongo_
+
+_$ python3.10 -m venv ./venv_
```
Comment (siriusdevs) [inline README.md:18]: implies i have to use python3.10? why? what if i dont? does not it run on 3.11?

File: README.md [hunk 3]
```diff
@@ -1 +1,96 @@
...
+
+_$ docker pull mongo_
+
+_$ git clone https://github.com/Max2288/rpm-hws_7-1_2023/_
+
+
+# How run
+Add .env file with your data
+
+Go to folder with project and run commands:
```
Comment (siriusdevs) [inline README.md]: nope, it wont run then. u need to go down repo/fastapi actually

File: README.md [hunk 4]
```diff
@@ -1 +1,96 @@
...
+_$ . ./venv/bin/activate_
+
+_$ pip install -r requirements.txt_
+
+# After start
+
+
+## anilist.co
+
+Example of data that we recieve:
```
Comment (siriusdevs) [inline README.md:29]: ANY of requests sent via postman (according to your suggestinons) return WARNING:  Invalid HTTP request received.

File: README.md [hunk 5]
```diff
@@ -1 +1,96 @@
...
+
+## anilist.co
+
+Example of data that we recieve: 
+    
+    {
+        "id": movie id,
+        "siteUrl": link to movie,
+        "title": {
+            "english": title's name on english,
```
Comment (siriusdevs) [inline README.md]: in english

File: README.md [hunk 6]
```diff
@@ -1 +1,96 @@
...
+# After start
+
+
+## anilist.co
+
+Example of data that we recieve: 
+    
+    {
+        "id": movie id,
+        "siteUrl": link to movie,
```
Comment (siriusdevs) [inline README.md:33]: could you please suggest real data though?
"id" : 1 and so on. users want to just copy your json, not make it up for themselves

File: README.md [hunk 7]
```diff
@@ -1 +1,96 @@
...
+* Go to Postman
+* Header example: _http://HOST:PORT/media?id=MOVIEIDTODELETE&auth=TOKEN_
+
+
+# .env
+## Below should be your data to project
+    HOST - app and database host
+    PORT - database host
+    DBNAME - database name
+    COLNAME - collection name
```
Comment (siriusdevs) [inline README.md:94]: what is this? do we need it though? works perfectly without it

File: README.md [hunk 8]
```diff
@@ -1 +1,96 @@
...
+You can delete media from page /media (Postman recommended)
+
+* Go to Postman
+* Header example: _http://HOST:PORT/media?id=MOVIEIDTODELETE&auth=TOKEN_
+
+
+# .env
+## Below should be your data to project
+    HOST - app and database host
+    PORT - database host
```
Comment (siriusdevs) [inline README.md]: host?

File: README.md [hunk 9]
```diff
@@ -1 +1,96 @@
...
+    DBNAME - database name
+    COLNAME - collection name
+    TOKEN - token to do REST API (should be - MAX228)
+    APP_PORT - application port
```
Comment (siriusdevs) [inline README.md]: added it for my dotenv, set it up for 8001. got server running on 8000 anyway

File: README.md [hunk 10]
```diff
@@ -1 +1,96 @@
...
+            },
+        "description": movie's description: str
+    }
+
+## DELETE request
+
+You can delete media from page /media (Postman recommended)
+
+* Go to Postman
+* Header example: _http://HOST:PORT/media?id=MOVIEIDTODELETE&auth=TOKEN_
```
Comment (siriusdevs) [inline README.md]: that is not a header example, it is a URI example. even if u meant query, they are parameters then

File: README.md [hunk 11]
```diff
@@ -1 +1,96 @@
-Keep your branches clean and nice. Use flake8 and tests. Good luck!
```
Comment (siriusdevs) [inline README.md:1]: where are the tests tho? 
u wont fool me
https://fastapi.tiangolo.com/tutorial/testing/

**Review by siriusdevs — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (Max2288) [reply README.md:18]: i recommend this settings, u can try yours

File: README.md (hunk 4 above)
Comment (Max2288) [reply README.md:29]: Give an example of your request

**Comment by siriusdevs** [comment]
1) post with id:999 creates next unused id (id:1 if there are no records yet)
that should not be a thing. you either post data the way it was given to you, either you do not do it and return another code - bad request, usually.
2) any delete request rerurns 200 OK "media deleted successfully". even if there were no such data. that it is not okay.

**Review by siriusdevs — CHANGES_REQUESTED** [review]
check your post and delete logic

**Comment by siriusdevs** [comment]
1) if user specifies nonexistent media in his delete request, you return 200 OK "Not found" crying_cat_face
2) Sometimes your server fails with internal error if user just reloads a "media list" page with browser. error output in file
[error.txt](https://github.com/siriusdevs/rpm-hws_7-1_2023/files/11034011/error.txt)

except for these two things, your project works like a charm

**Comment by siriusdevs** [comment]
you have 31+4 errors ignored by setup.cfg, agreed amount was 30

**Pull request merged**